mspray/apps/main/tests/views/test_performance.py

```diff
@@ -34,3 +43,223 @@ def test_team_leader_performance(self):
         response = view(request, slug=sprayarea.id)
         self.assertEqual(response.status_code, 200)
         self.assertTrue('Supervisor' in str(response.render().content))
+
+    def test_mda_spray_operator_summary_view(self):
+        """Test MDASprayOperatorSummaryView.
+
+        This is done by confirming the data received in the response
+        context is being aggregated appropriately within the view
+        """
+        # first we load our test data
+        data_setup()
+        self._load_fixtures()
+        rhc = Location.objects.get(name='Chadiza_104')
+        spray_operator = SprayOperator.objects.first()
+        team_leader = TeamLeader.objects.first()
+        # next we identify a SprayDay object for the spray operator
+        spray_operator.team_leader = team_leader
+        spray_operator.rhc = rhc
+        spray_operator.save()
+
+        # update spray day object for the particular spray operator
+        spray_day = SprayDay.objects.filter(spray_operator=spray_operator)
+        spray_day.update(sprayable=True)
+
+        # Create performance report objects from the submissions made
+        # by the particular sprayoperator.
+
+        performance_report(spray_operator)
+        report1 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report1.found = 7
+        report1.reported_sprayed = 16
+        report1.reported_found = 22
+        report1.save()
+
+        report2 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report2.id = None
+        report2.sprayformid = 7658
+        report2.refused = 6
+        report2.found = 12
+        report2.reported_sprayed = 6
+        report2.save()
+
+        factory = RequestFactory()
+        request = factory.get("spray-operators/2/summary")
+        view = MDASprayOperatorSummaryView.as_view()
+
+        response = view(request, rhc_id=rhc.id)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.context_data['rhc_name'], "Chadiza_104")
+
+        # Query obtains all data for the SprayOperator
+        # including submissions made and passes this to the serializer
+
+        queryset = SprayOperator.objects.raw(
+            MDA_SOP_PERFORMANCE_SQL, [rhc.id]
+        )
+        serializer = SprayOperatorPerformanceReportSerializer(
+            queryset, many=True
+        )
+
+        result = {
+            'other': 0,
+            'refused': 6,
+            'sprayed': 2,
+            'sprayable': 19,
+            'not_sprayable': 0,
+            'not_sprayed_total': 6,
+            'data_quality_check': False,
```

Comment:
Why is the data quality check `False` in this case?

Reply:
@moshthepitt the data quality check is not a computated value from the serializer, rather it is picked from the last performance report object submitted by the spray_operator(distinguished by the spray_date).
Thus, since when creating the test case performance report objects I did not specify the boolean value to the 'True'...it was set to by default to be false, and eventually the output from the serializer was False.